core/blockchain.go

```diff
@@ -2336,24 +2336,25 @@ func (bc *BlockChain) UpdateValidatorVotingPower(
 				return nil, err
 			}
 
-			aprComputed, err := apr.ComputeForValidator(
-				bc, block, wrapper,
-			)
-			if err == nil && aprComputed != nil {
-				a := *aprComputed
+			stats.APR = numeric.ZeroDec()
+			if wrapper.TotalDelegation().Cmp(common.Big0) > 0 {
+				aprComputed, err := apr.ComputeForValidator(
+					bc, block, wrapper,
+				)
+				if err != nil {
+					if err == apr.ErrInsufficientEpoch {
```

Comment:
but now its wrapped right, so need the .Contain, and in the log, utils.Logger().Err(err).Info().Msg("..")

Reply:
the err being on the plain err, not the contained one

Reply:
oops.

Reply:
`errors.Cause(` ?

Reply:
Ya, like you used in another place before, I think ya, .Cause (I rememberde something with C)

Reply:
if we use .Err(err) and don't return, sounds weird... ?

Reply:
No, you're not understanidng the logger API (I was confused by this as well), 

It would be weird if we had 

utils.Logger().Error().Err(err).Msg()

and didn't return, 

but this is `.Info()`

Reply:
we're aware that there is an error but the business logic handles it as an acceptable error. Error is just a value and we can branch on its value as well (although usually shouldn't)